src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/ExperimentalAttribute.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>
+    ///  Indicates that an API is experimental and it may change in the future.
+    /// </summary>
+    /// <remarks>
+    ///   This attribute allows call sites to be flagged with a diagnostic that indicates that an experimental
+    ///   feature is used. Authors can use this attribute to ship preview features in their assemblies.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Assembly |
```

Comment:
@joperezr @stephentoub Let's not include "Assembly" and "Module" in the list. I don't think those targets are valid for other attributes of this kind (`Obsolete`, `Windows.Foundation.Metadata.Experimental`, `
Windows.Foundation.Metadata.Deprecated`, ...).

Reply:
cc: @terrajobst @geeknoid 

Thanks for the feedback @jcouv. Aside from the fact that other attributes don't support this, is there a particular reason why you don't think we should allow Assembly or Module use? Seems to me that there are valid scenarios where people developing new assemblies might opt to use it at that level, in the past this has been done mostly via using prerelease package versions but I think that not everybody uses that and there are some scenarios where it might be better to have stable p ackages with assemblies marked as experimental in them.

Reply:
The main reason for my question is precedent. The secondary reason is implementation cost (deviating from existing attributes means additional complexity and testing).
I'm not saying it's a bad idea, but that this previous design was worked, I think we'd need a strong reason to revise it (why was it sufficient for those other attributes but not this one).

Reply:
fair enough. @bartonjs @terrajobst @stephentoub Do you know off the top of your head why the Obsolete attribute doesn't support Assembly or Module targets? I would have expected the same arguments should be made for both Obsolete and Experimental, since they are essentially the same attribute but just tracking the two opposing ends of an API lifecycle.

Reply:
I believe R9 has assembly-level applications of the attribute. E.g., https://github.com/dotnet/extensions/blob/d558f8de86adf93044bc5386bd4367b3d509ead3/Directory.Build.targets#L68-L72

Reply:
For the platform compat attributes we allowed assembly/module and we use it quite a bite.

Reply:
I'm fine with excluding them for now if we want to match ObsoleteAttribute. We can revisit then we run into limitations.

Reply:
@geeknoid FYI

Reply:
FYI, the compiler/roslyn change was merged.